Dark SUSY and Scouting di-muon paths

gomber opened the pull request:
I have created Prompt Monitoring code for the Dark SUSY and Scouting di-muon paths.
Correspond to the following JIRA tickets : 
https://its.cern.ch/jira/browse/CMSHLT-1244
https://its.cern.ch/jira/browse/CMSHLT-1243

Testing the code locally!

cmsbuild commented:
A new Pull Request was created by  @gomber  for master.

It involves the following packages:

DQMOffline/Trigger


@cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mtosi commented on DQMOffline/Trigger/plugins/MuonMonitor.cc:
```diff
@@ -0,0 +1,288 @@
+  setTitle(muonPhiME_,"Muon #phi","events / 0.1 rad");
+
+
+  histname = "muon_eta"; histtitle = "Muon eta";
```
please add eta variable binning

mtosi replied:
isn't this analysis interested in di-muon mass as well ?
why don't you add it ?

mtosi replied:
in addition, shouldn't you check distributions like dxy, dz, lxy, etc ?

mtosi replied:
why can't you use the already available code by MUO POG ?

gomber replied:
ok will do now.

mtosi commented on DQMOffline/Trigger/plugins/MuonMonitor.cc:
```diff
@@ -0,0 +1,288 @@
+  //  float phi = pfmet.phi();
+
+
+  edm::Handle<reco::MuonCollection> muonHandle;
```
uhmmm, I'm confused :(
I thought the scouting analyses do not rely on offline reco
=> I'm surprised by this collection :(

gomber replied:
yes but we need to make efficiency as a function of muon pt, thats why.
Also I have other trigger paths included in this PR which depends on muon.

gomber replied:
ok I confirmed with Swagata and she told me for scouting trigger, they would like to see efficiency as a function of hlt object pt. I will add that information.

mtosi commented on DQMOffline/Trigger/python/ExoticaMonitoring_Client_cff.py:
```diff
@@ -16,6 +16,26 @@
 
+
+muonEfficiency = DQMEDHarvester("DQMGenericClient",
+    subDirs        = cms.untracked.vstring("HLT/Muon/*"),
```
uhmm, shouldn't we have an ad hoc directory for this ?

gomber replied:
I don't follow this comment Mia, so for other Muon paths, they do the efficiency using tag and probe method, and for these paths I use orthogonal method, so thats why I created new code .

mtosi commented on DQMOffline/Trigger/python/MuonMonitor_cff.py:
```diff
@@ -0,0 +1,36 @@
+
+DST_DoubleMu3_noVtx_CaloScouting_monitoring = hltMuonmonitoring.clone()
+DST_DoubleMu3_noVtx_CaloScouting_monitoring.FolderName = cms.string('HLT/Muon/DST_DoubleMu3_noVtx_CaloScouting/')
+DST_DoubleMu3_noVtx_CaloScouting_monitoring.numGenericTriggerEventPSet.hltPaths = cms.string("DST_DoubleMu3_noVtx_CaloScouting_v*")
```
I do not see this path in the STORM google doc
please, make sure the corresponding JIRA points to this PR
thanks

gomber replied:
yes this point is not yet integrated in STORM as swagata is completing the homework, but I still went ahead and did prompt monitoring. I will make sure about JIRA>

swagata87 replied:
In case it is useful, this is the JIRA for the scouting dimuon path : https://its.cern.ch/jira/browse/CMSHLT-1243

mtosi commented on DQMOffline/Trigger/python/MuonMonitor_cff.py:
```diff
@@ -0,0 +1,36 @@
+DST_DoubleMu3_noVtx_CaloScouting_monitoring.FolderName = cms.string('HLT/Muon/DST_DoubleMu3_noVtx_CaloScouting/')
+DST_DoubleMu3_noVtx_CaloScouting_monitoring.numGenericTriggerEventPSet.hltPaths = cms.string("DST_DoubleMu3_noVtx_CaloScouting_v*")
+DST_DoubleMu3_noVtx_CaloScouting_monitoring.denGenericTriggerEventPSet.andOrL1 = cms.bool( True )
+DST_DoubleMu3_noVtx_CaloScouting_monitoring.denGenericTriggerEventPSet.l1Algorithms = cms.string("L1_DoubleMu0er1p6_dEta_Max1p8_OS_v*", "L1_DoubleMu_11_4_v*", "L1_DoubleMu_12_5_v*","L1_DoubleMu_13_6_v*","L1_TripleMu0_v*","L1_QuadMu0_v*")
```
is it working ?

mtosi commented on DQMOffline/Trigger/python/MuonMonitor_cfi.py:
```diff
@@ -0,0 +1,34 @@
+
+hltMuonmonitoring.denGenericTriggerEventPSet.andOr         = cms.bool( False )
+hltMuonmonitoring.denGenericTriggerEventPSet.andOrHlt        = cms.bool( True )
+hltMuonmonitoring.denGenericTriggerEventPSet.hltInputTag   = cms.InputTag( "TriggerResults::HLT" )
```
I would suggest to drop the HLT part ...

mtosi commented:
tracked by #19142

mtosi commented:
ah, what about HLT_TrkMu15_DoubleTrkMu5NoFiltersNoVtx ?
I think I do not see it here, while it is in the STORM google doc

gomber commented:
@dildick hey do we need this path "HLT_TrkMu15_DoubleTrkMu5NoFiltersNoVtx ", from JIRA I see only 2 paths which I added to prompt monitoring, please confirm.

dildick commented:
The two trigger paths that we have in mind are currently: HLT_TrkMu16_DoubleTrkMu6NoFiltersNoVtx (signal) and HLT_TrkMu17_DoubleTrkMu8NoFiltersNoVtx (backup). I am trying to sort out today whether the signal trigger has high enough cuts. If not, it would likely be HLT_TrkMu16_DoubleTrkMu7NoFiltersNoVtx. But HLT_TrkMu15_DoubleTrkMu5NoFiltersNoVtx is not expected to be in the menu

gomber commented:
@dildick please let me know whenever you finalize it. And regarding efficiency measurement techqniue, I have talked to Mia and we think we need to be a bit careful with it, as at present you asked me to use orthogonal method using met dataset (which could be biased as we are looking st muon paths). So how about we use reference method, where we measure efficiency wrt to other   path with lower threshold which you can also ask for with some prescale? What do you think? Can you prepare another backup path with lower thresholds on muons pt with some prescale?

dildick commented:
So for our trigger efficiency studies we excluded events in the MET dataset which were triggered by MET + muon or other non-pure MET triggers. That should largely remove the bias. I'm not sure what bias you are thinking of. In any case, I'd rather not change the method so late in the process.

mtosi commented:
so, 
1.
these veto has to be applied in the DQM as well
2. an estimation of the needed stat for getting meaningful monitoring has to be provided
3.
this thoughts and design was supposed to be done well in advance, and of course even the DQM code development was expected to be done in advance as well (around march/ april to be more precise)

the requirement of having a prompt monitoring in place does not mean having some sort of code which produces some sort of plots w/ some sort of statistics

i'm kind of tired / annoyed / 'stufa' of such kind of approach like yours !

> On 9 Jun 2017, at 19:50, Sven Dildick <notifications@github.com> wrote:
> 
> So for our trigger efficiency studies we excluded events in the MET dataset which were triggered by MET + muon or other non-pure MET triggers. That should largely remove the bias. I'm not sure what bias you are thinking of. In any case, I'd rather not change the method so late in the process.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

fwyzard commented:
> In any case, I'd rather not change the method so late in the process.

Deadline's still June 20th. Of course there will be an other "train" after that...

cmsbuild commented:
Pull request #19154 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
Hi All, 

So I have changed the method for dark SUSY triggers, i.e we use reference method now using the backup path which Sven developed and he will be presenting in TSG today. In addition, also implemented some of the Mia comments, only thing left is adding dxy  and dz muon variables, Since this code runs over reco collection (i am not muon expert), I am not able to get it right, if @dildick knows how to do that please let me know. 

Regarding Scouting paths, @swagata87 please correct me if I am wrong, but I guess this code will not run as the collection "ScoutingMuonCollection" doesn't exists yet? So what do we do?

cmsbuild commented:
Pull request #19154 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
@mtosi, everyone
I have updated all the comments for dark SUSY, also implemented comments from david lange which i received on 18971 related to code. 

Regarding scouting paths, I think @swagata87 is discussing with Mia, so we will follow that up once it is resolved. 

In the mean time, I think we can go ahead and test this @dmitrijus

mtosi commented:
yes, please
(drop any remnants related to the scouting paths, please)

On Wed, Jun 14, 2017 at 5:24 PM, gomber <notifications@github.com> wrote:

> @mtosi <https://github.com/mtosi>, everyone
> I have updated all the comments for dark SUSY, also implemented comments
> from david lange which i received on 18971 related to code.
>
> Regarding scouting paths, I think @swagata87
> <https://github.com/swagata87> is discussing with Mia, so we will follow
> that up once it is resolved.
>
> In the mean time, I think we can go ahead and test this @dmitrijus
> <https://github.com/dmitrijus>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19154#issuecomment-308466852>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt584iCmeWOTgILNRj7mGdY24PPn6Qjks5sD_tLgaJpZM4N0sxS>
> .
>

lpernie commented:
Hi @fwyzard, do you know when the next deadline (after June 20) will be?

fwyzard commented:
I would say around mid-July, for example on July 12th.

cmsbuild commented:
Pull request #19154 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
@mtosi, I have removed scouting paths from this PR an committed. I am waiting on Dark SUSY to finalize the trigger efficiency measurement method, then we can start testing the code.

cmsbuild commented:
Pull request #19154 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #19154 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #19154 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

gomber commented:
@dmitrijus I have implemented all the comments for this PR, also added the efficiency measurement for the reference path which was one of the action item.

Could you please test the code?

Regards
Bhawna

dmitrijus commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 74f80c3fb1cdb1b810675fc8b3f0dc076b052612
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19154/20877/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5ad2ca330590b7d14a64a053f7f4350de7357460
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19154/20877/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19154/20877/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19154/20877/summary.html

Comparison Summary:
 - You potentially added 35 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1802916
 - DQMHistoTests: Total failures: 29339
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1773411
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

mtosi commented:
please, rebase

mtosi commented on DQMOffline/Trigger/python/MuonMonitor_cff.py:
```diff
@@ -0,0 +1,34 @@
+
+TrkMu12_DoubleTrkMu5NoFiltersNoVtx_monitoring = hltMuonmonitoring.clone()
+TrkMu12_DoubleTrkMu5NoFiltersNoVtx_monitoring.FolderName = cms.string('HLT/Exotica/TrkMu12_DoubleTrkMu5NoFiltersNoVtx/')
+TrkMu12_DoubleTrkMu5NoFiltersNoVtx_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_TrkMu12_DoubleTrkMu5NoFiltersNoVtx_v*")
```
I do not see this path, but 
HLT_TrkMu15_DoubleTrkMu5NoFiltersNoVtx_v* seems to not be covered, instead

gomber commented:
@mtosi In the JIRA it is clearly mentioned by the analyzers tthat they want to add HLT_TrkMu12... path and want to drop HLT_TrkMu15_DoubleTrkMu5NoFiltersNoVtx_v.

gomber commented:
@mtosi @dmitrijus I see these conflicting files, because everyone used the same e.g. #19293 also get this issue of conflicts? So what do we do in this case?

mtosi commented:
in the silvio's google doc is not

On Mon, Jun 26, 2017 at 3:19 PM, gomber <notifications@github.com> wrote:

> @mtosi <https://github.com/mtosi> In the JIRA it is clearly mentioned by
> the analyzers tthat they want to add HLT_TrkMu12... path and want to drop
> HLT_TrkMu15_DoubleTrkMu5NoFiltersNoVtx_v.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19154#issuecomment-311056889>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58wrDt1-h2E-v3XKrKtl4QZ0iQs9Hks5sH6_wgaJpZM4N0sxS>
> .
>

gomber commented:
@mtosi yes you can the see the discussion and issues which Sven is having with the integration and rates test : https://its.cern.ch/jira/browse/CMSHLT-1244 Thats why the path is not included in the Sulvio google doc, but once @dildick solves the issues then the updated path will be included. I updated this PR with the paths which we want to put online.

mtosi commented:
thanks for clarifying it

> On 26 Jun 2017, at 15:37, gomber <notifications@github.com> wrote:
> 
> @mtosi yes you can the see the discussion and issues which Sven is having with the integration and rates test : https://its.cern.ch/jira/browse/CMSHLT-1244 Thats why the path is not included in the Sulvio google doc, but once @dildick solves the issues then the updated path will be included. I updated this PR with the paths which we want to put online.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
Pull request #19154 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #19154 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #19154 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #19154 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #19154 was updated. @ghellwig, @perrozzi, @ianna, @kpedro88, @fabozzi, @Martin-Grunewald, @mommsen, @vanbesien, @govoni, @perrotta, @civanch, @monttj, @silviodonato, @cmsbuild, @fwyzard, @GurpreetSinghChahal, @davidlange6, @smuzaffar, @Dr15Jones, @kkousour, @mdhildreth, @cerminar, @slava77, @ggovi, @mulhearn, @efeyazgan, @vazzolini, @rekovic, @kmaeshima, @arunhep, @emeschi, @thuer, @dmitrijus, @alja, @franzoni, @lpernie can you please check and sign again.

gomber commented:
@mtosi I have implemented the strategy for muon+photon paths as we discussed in the morning, So all is done from my side for these paths too 
HLT_DoubleMu43NoFiltersNoVtx_v1
HLT_DoubleMu48NoFiltersNoVtx_v1 (backup path)
HLT_Mu43NoFiltersNoVtx_Photon43_CaloIdL_v1
HLT_Mu48NoFiltersNoVtx_Photon48_CaloIdL_v1 (backup path)

Please have a look and sign off.

kpedro88 commented:
-1
never merge, always rebase
http://cms-sw.github.io/tutorial-resolve-conflicts.html

gomber commented:
@kpedro88 I started with rebase, but then while compiling the code never compiles and give error related to things which I haven't touched, I wanted to complete the code so started with the merge.

kpedro88 commented:
@gomber did you follow the rebase instructions that I linked? (again, merging is never the correct way to update a PR or resolve conflicts)

slava77 commented:
-1
needs rebase or remake to keep only relevant commits

gomber commented:
@kpedro88 i tried your instricuons with the new CMSSW release, here is what I followed 
cmsrel CMSSW_9_3_X_2017-06-30-1100
cd CMSSW_9_3_X_2017-06-30-1100/src/
cmsenv
git cms-init
git cms-checkout-topic -u gomber:ExoTrigger_MuonMonitoring92X
scram b -j 8
git rebase --onto CMSSW_9_3_X CMSSW_9_2_X ExoTrigger_MuonMonitoring92X

But this command gives me error 
[lxplus092@src] git rebase --onto CMSSW_9_3_X CMSSW_9_2_X ExoTrigger_MuonMonitoring92X
fatal: Needed a single revision
invalid upstream CMSSW_9_2_X

Any suggestions? Otherwise I will open a new PR and close this one.

gomber commented:
@kpedro88  Ok Guys, I don't know what happened here! I was trying to rebase this branch following the instructions and looks like i did force commit and everything is gone. I even need to rewrite some part of the code as it is not even showing it in my own repository. I will do later today and open a new PR. If someone knows how to get back the code, please let me know! I will appreciate it.

kpedro88 commented:
@gomber CMSSW_9_2_X and CMSSW_9_3_X are branches, not releases. You need to use release tags or commit hashes (e.g. CMSSW_9_3_X_2017-06-30-1100).

If you still have some of the working areas where you were trying merge or rebase, you can recover the previous state of the branch from `git reflog`. But it looks like you already remade the PR.
